stable/database/files/nuosm

```diff
@@ -761,9 +761,10 @@ else
 
   wotIdid=""
 
-  # resurrect my archive if it is removed
-  # this will likely never execute but is kept to be sure
-  resurrectRemovedArchive
+  # resurrect my archive only if info.json file is available, otherwise
+  # nuodocker will create a new one; this is needed because KAA will remove
+  # archives for scaled down SM statefulsets
+  [ -f "$DB_DIR/info.json" ] && resurrectRemovedArchive
```

Comment:
I just looked at the code for `resurrectRemovedArchive` and it is doing screen-scraping of `nuocmd show archives` and filtering on the hostname to obtain the archive ID, which is readily available in `info.json`. I think we should just update `resurrectRemovedArchive` to get the archive ID from `info.json`.

Reply:
Good call! Yes, we need to fix the `resurrectRemovedArchive` so that it uses the already available `$myArchive` variable instead of fetching the re-fetching the archived. I added another commit to address that.

@acabrele reminded me that this "screen-scraping" won't work if the SM pod was previously in _crashloop_ state having some startup problem causing the admin to display `<UNKNOWN ADDRESS>` since the engine never passed `CONFIGURED` state. This won't happen when NuoDB 4.2+ is used, however, this change in `nuosm` should fix it when using previous versions of the product.